src/irmin-pack/s.ml

```diff
@@ -55,41 +55,72 @@ module type Specifics = sig
+        running *)
+
+    (** {2 Low-level API} *)
+    val start_exn :
+      ?unlink:bool -> throttle:throttle -> repo -> commit_key -> bool Lwt.t
+    (** [start_exn] tries to start the gc process and returns true if the gc is
+        launched.
+
+        If [unlink] is false then temporary files and files from the previous
+        generation will be kept on disk after the gc finished. This option is
+        useful for debugging. The default is true.
 
-      If [unlink] is false then temporary files and files from the previous
-      generation will be kept on disk after the gc finished. This option is
-      useful for debugging. The default is true.
+        If [throttle] is [Skip] and there is a concurrent gc, [start_gc] returns
+        false immediately without launching a second gc. If [throttle] is
+        [Block] and there is a concurrent gc, [start_gc] blocks waiting for the
+        previous gc to finish and only after launches a second gc. If the
+        previous GC failed, the function returns without launching a new GC.
 
-      If [throttle] is [Skip] and there is a concurrent gc, [start_gc] returns
-      false immediately without launching a second gc. If [throttle] is [Block]
-      and there is a concurrent gc, [start_gc] blocks waiting for the previous
-      gc to finish and only after launches a second gc. If the previous GC
-      failed, the function returns without launching a new GC.
+        TODO: Detail exceptions raised. *)
 
-      TODO: Detail exceptions raised. *)
+    val finalise_exn : ?wait:bool -> repo -> bool Lwt.t
+    (** [finalise_exn ?wait repo] waits for the gc process to finish in order to
+        finalise it. It returns true if a GC was finalised.
 
-  val finalise_gc : ?wait:bool -> repo -> bool Lwt.t
-  (** [finalise_gc ?wait repo] waits for the gc process to finish in order to
-      finalise it. It returns true if a GC was finalised.
+        Finalising consists of mutating [repo] so that it points to the new file
+        and to flush the internal caches that could be referencing GCed objects.
 
-      Finalising consists of mutating [repo] so that it points to the new file
-      and to flush the internal caches that could be referencing GCed objects.
+        If [wait = true] (the default), the call blocks until the GC process
+        finishes. If [wait = false] it either returns [false], raises an
+        exception or finalises and returns [true].
 
-      If [wait = true] (the default), the call blocks until the GC process
-      finishes. If [wait = false] it either returns [false], raises an exception
-      or finalises and returns [true].
+        If there are no running gcs, the call is a no-op and it returns false.
 
-      If there are no running gcs, the call is a no-op and it returns false.
+        TODO: Detail exceptions raised. *)
 
-      TODO: Detail exceptions raised. *)
+    (** {2 High-level API} *)
+
+    val run : repo -> commit_key -> bool Lwt.t
```

Comment:
The final boolean is not really clear here; as we are running the GC in blocking mode, the boolean is false only if they were an error - in that case, wouldn't it be better to return an abstract (but pretty-printable error) instead?

I would also return a structured result, returning a few stats (like the total time taken by the GC). That'll allow client users to pretty-print it using their own logging/formatter facilities.